qdrant_client/common/client_warnings.py

```diff
@@ -0,0 +1,21 @@
+import warnings
+from typing import Optional
+
+SEEN_MESSAGES = set()
+
+
+def show_warning(message: str, category: type[Warning] = UserWarning) -> None:
+    warnings.warn(message, category, stacklevel=2)
+
+
+def show_warning_once(
+    message: str, category: type[Warning] = UserWarning, idx: Optional[str] = None
+) -> None:
+    """
+    Show a warning of the specified category only once per program run.
+    """
+    key = idx if idx is not None else message
+
+    if key not in SEEN_MESSAGES:
+        SEEN_MESSAGES.add(key)
+        show_warning(message, category)
```

Comment:
```suggestion
        show_warning(message, category)
        
def deprecation_warning(message: str) -> None:
    show_warning(message, DeprecationWarning)


def deprecation_warning_once(message: str, idx: Optional[str]) -> None:
    show_warning_once(message, DeprecationWarning, idx)
```

Changes of unrelated methods could be avoided if you just add those